docs/site/Interceptors.md

```diff
@@ -542,6 +544,75 @@ export class InvocationContext extends Context {
 It's possible for an interceptor to mutate items in the `args` array to pass in
 transformed input to downstream interceptors and the target method.
 
+### Source for an invocation
+
+The `source` property of `InvocationContext` is defined as `InvocationSource`:
+
+```ts
+/**
+ * An interface to represent the caller of the invocation
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export interface InvocationSource<T = any> {
```

Comment:
Can we use `unknown` instead of `any` please? Or perhaps `object`?

Reply:
> Can we use `unknown` instead of `any` please? Or perhaps `object`?

I feel this thread is not resolved yet.

See also https://github.com/strongloop/loopback-next/pull/4168#discussion_r349093431.

My main concerns is that we make it easy to introduce `any` value into code written by consumers of this interface. Once the `any` type is introduced, it pretty much disables all type checks going forward. I am ok to allow to use our users `any`, but I want to make that fact obvious in their code.

Consider the following code:

```ts
const foo: Interceptor = async (invocationCtx, next) => {
  const source = invocationCtx.source;
  const bindingPath = source.getBindingPath();
  // etc.
};
```

The code looks legitimate, it will compile (because `source` is `any`), it will even work as long as the interceptor is invoked only for proxies. The problem will manifest when the interceptor is called for a controller/handler route. Notice that `bindingPath` has also type `any`, which is super easy to miss while reading the code.

Reply:
Oh. I see that in the code, you have already changed the default `T` value to `unknown`. 

Please fix the documentation too and then this thread will be resolved 👍